Decred Wallet/Features/Send/SendViewController.swift

```diff
@@ -503,33 +503,37 @@ extension SendViewController {
     }
     
     func checkAddressFromQrCode(textScannedFromQRCode: String?) {
-        guard var capturedText = textScannedFromQRCode else {
+        
+        guard let capturedText = textScannedFromQRCode else {
             self.destinationAddressTextField.text = ""
             return
         }
+        let URI: DecredAddressURI = DecredAddressURI(uriString: capturedText)
         
-        if capturedText.starts(with: "decred:") {
-            capturedText = capturedText.replacingOccurrences(of: "decred:", with: "")
+        guard let address = URI.address else {
+            return
         }
         
-        if capturedText.count < 25 {
+        self.dcrAmountTextField.text = String(format: "%", URI.amount ?? "")
```

Comment:
Shouldn't displayed the parsed amount until after the address is validated. We wouldn't want to display only the amount and ignore the address field.

Also, no need to clear the amount text field (setting the text to "") if there was no amount in the scanned text. Should just leave the amount field as it was.

Reply:
resolved in `10b8349`